sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -289,9 +289,11 @@ export function appendQueryParams(
     if (typeof value === "string") {
       searchPieces.push(`${name}=${value}`);
     } else {
-      // QUIRK: If we get an array of values, include multiple key/value pairs
-      for (const subValue of value) {
-        searchPieces.push(`${name}=${subValue}`);
+      if (value) {
```

Comment:
The type for `value` is either `string | string[]` so I think the function is correct in handling it as such only. Could you check which caller puts an undefined in a map that is supposed to hold `string | string[]` only?

Reply:
@deyaaeldeen If you look at the combinedParams object it will look as:
```
{
  'api-version' => '2021-07-01-preview',
  'operationSource' => 'other',
  'asyncinfo' => undefined
}
```
As you can see, the `asyncinfo` is `undefined` which is causing this issue. The URL looks like `https://management.azure.com/subscriptions/3755ce94-6c61-4b5c-b0b1-b2ce86fc540f/providers/Microsoft.Devices/locations/eastus/operationResults/aWQ9b3NfaWhfNWZiNWU3OTEtZGJmOS00ZjFiLThkY2EtYzBiOTJmMTI2MjIxO3JlZ2lvbj1lYXN0dXM=?api-version=2021-07-01-preview&operationSource=other&asyncinfo`

Reply:
Ops sorry, I was confused. I thought the value came from `queryParams` map not from `combinedParams`. Ok, I looked at `combinedParams` type and it is `Map<string, string | string[]>` which should not still include `undefined` in it. Since `simpleParseQueryParams` is the one the produce it, so it is the one that needs to be updated. It splits each param on `=` as follows: `const [name, value] = pair.split("=", 2);` without validating that `value` actually is defined event though the type does not allow for `undefined`. There're two options here, either relax the type to allow for `undefined` or validate `value` to always be defined. I think the former option makes sense here because we would like to preserve whatever is in the URL but I could be wrong. @xirzec any thoughts on this?

Reply:
If the value is undefined, shouldn't we just push `${name}=`?

Reply:
I'm a little torn. I want to say "of course nobody would want a query param with no value!" but as soon as I say that I worry some service of ours indeed would expect that.

I'm leaning towards doing Jeremy's suggestion, but in that case to Deya's point we should update the map to include `| undefined` as one of the possible values.

Reply:
> If the value is undefined, shouldn't we just push ${name}=?

The original URL in [Sarangan's example](https://github.com/Azure/azure-sdk-for-js/pull/18621#discussion_r746350333) does not have the `=` sign, so I think we will have to preserve this which means distinguishing between `value` being undefined and being an empty string.

Reply:
Ok. I have changed the code to include undefined as a possible return value during parsing and changed the code